### PR TITLE
Adding RestActions support Create/Update Detector API

### PR DESCRIPTION
**Pull request by saratvemulapalli**
*Issue #195 *

*Description of changes:*
Adding RestActions support Create/Update Detector API which is needed to support AD FGAC

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -254,7 +254,8 @@ List<String> jacocoExclusions = [
         'com.amazon.opendistroforelasticsearch.ad.transport.StatsAnomalyDetectorTransportAction',
         'com.amazon.opendistroforelasticsearch.ad.transport.DeleteAnomalyDetectorTransportAction*',
         'com.amazon.opendistroforelasticsearch.ad.transport.GetAnomalyDetectorTransportAction*',
-        'com.amazon.opendistroforelasticsearch.ad.transport.GetAnomalyDetectorResponse'
+        'com.amazon.opendistroforelasticsearch.ad.transport.GetAnomalyDetectorResponse',
+        'com.amazon.opendistroforelasticsearch.ad.transport.IndexAnomalyDetectorRequest'
```
Comment (saratvemulapalli) [inline build.gradle:258]: Added IndexAnomalyDetectorRequest using Powermock but JaCoCo doesnt read lines covered by Powermock tests.

File: src/main/java/com/amazon/opendistroforelasticsearch/ad/rest/handler/AnomalyDetectorActionHandler.java [hunk 2]
```diff
@@ -49,47 +49,57 @@
      * @param clusterService ES cluster service
      * @param client ES node client
      * @param detectorId detector identifier
-     * @param channel ES rest channel
+     * @param listener Listener to send response
      * @param function AD function
      */
```
Comment (ylwu-amzn) [inline src/main/java/com/amazon/opendistroforelasticsearch/ad/rest/handler/AnomalyDetectorActionHandler.java:54]: Add `xContentRegistry` ?

Comment (saratvemulapalli) [reply src/main/java/com/amazon/opendistroforelasticsearch/ad/rest/handler/AnomalyDetectorActionHandler.java:54]: Nice catch. 
Sure will fix them.

File: src/main/java/com/amazon/opendistroforelasticsearch/ad/rest/handler/IndexAnomalyDetectorActionHandler.java [hunk 3]
```diff
@@ -132,11 +141,15 @@ public IndexAnomalyDetectorActionHandler(
      */
     public void start() throws IOException {
         if (!anomalyDetectionIndices.doesAnomalyDetectorIndexExist()) {
+            System.out.println("AnomalyDetector Indices do not exist");
```
Comment (ylwu-amzn) [inline src/main/java/com/amazon/opendistroforelasticsearch/ad/rest/handler/IndexAnomalyDetectorActionHandler.java:144]: Remove System.out.println

Comment (saratvemulapalli) [reply src/main/java/com/amazon/opendistroforelasticsearch/ad/rest/handler/IndexAnomalyDetectorActionHandler.java:144]: Sure.. there are other debugging logs as well. Will remove them.

File: src/main/java/com/amazon/opendistroforelasticsearch/ad/rest/handler/IndexAnomalyDetectorActionHandler.java [hunk 4]
```diff
@@ -183,28 +200,35 @@ private void onGetAnomalyDetectorResponse(GetResponse response) throws IOExcepti
...
             logger.error(errorMsg);
-            onFailure(new IllegalArgumentException(errorMsg));
+            listener.onFailure(new IllegalArgumentException(errorMsg));
         } else {
             searchAdInputIndices(null);
         }
     }
 
     private void searchAdInputIndices(String detectorId) {
+        logger.info("#222 searchAdInputIndices is called ");
```
Comment (ylwu-amzn) [inline src/main/java/com/amazon/opendistroforelasticsearch/ad/rest/handler/IndexAnomalyDetectorActionHandler.java:231]: This log is for debugging? If yes, change to debug level or remove it.

**Comment by ohltyler** [comment]
Could you rebase with master to only have the changes related to create/update detector?

File: src/main/java/com/amazon/opendistroforelasticsearch/ad/transport/GetAnomalyDetectorTransportAction.java [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+import com.amazon.opendistroforelasticsearch.ad.model.DetectorProfile;
+import com.amazon.opendistroforelasticsearch.ad.model.ProfileName;
+import com.amazon.opendistroforelasticsearch.ad.settings.AnomalyDetectorSettings;
+import com.amazon.opendistroforelasticsearch.ad.util.DiscoveryNodeFilterer;
+import com.amazon.opendistroforelasticsearch.ad.util.RestHandlerUtils;
+import com.google.common.collect.Sets;
+
+public class GetAnomalyDetectorTransportAction extends HandledTransportAction<GetAnomalyDetectorRequest, GetAnomalyDetectorResponse> {
+
+    private static final Logger LOG = LogManager.getLogger(StopDetectorTransportAction.class);
```
Comment (ylwu-amzn) [inline src/main/java/com/amazon/opendistroforelasticsearch/ad/transport/GetAnomalyDetectorTransportAction.java:62]: Change logger name to current class `GetAnomalyDetectorTransportAction.class`

File: src/main/java/com/amazon/opendistroforelasticsearch/ad/rest/handler/IndexAnomalyDetectorActionHandler.java (hunk 4 above)
Comment (saratvemulapalli) [reply src/main/java/com/amazon/opendistroforelasticsearch/ad/rest/handler/IndexAnomalyDetectorActionHandler.java]: Sure yeah it was for debugging. I removed them : )

**Comment by saratvemulapalli** [comment]
> Could you rebase with master to only have the changes related to create/update detector?

Sure, just rebased and merged master changes into this PR.

File: src/main/java/com/amazon/opendistroforelasticsearch/ad/transport/GetAnomalyDetectorTransportAction.java (hunk 5 above)
Comment (saratvemulapalli) [reply src/main/java/com/amazon/opendistroforelasticsearch/ad/transport/GetAnomalyDetectorTransportAction.java]: Nice catch. Sure fixed it.

**Review by ohltyler — APPROVED** [review]
LGTM!

**Review by ylwu-amzn — APPROVED** [review]
LGTM. Thanks for the change!

**Pull request merged**